packages/client/src/pages/Store/index.tsx

```diff
@@ -0,0 +1,286 @@
+import { Header } from "components/Header";
+import { useApi } from "context/api";
+import { IApiError, IDiscount, IGameCreator, IGenre } from "interfaces/api";
+import { IGameForUI } from "interfaces/app";
+import React, { useEffect, useRef, useState } from "react";
+import { filterGames, getFilterOptions, sortGames } from "utils/api";
+import { FaWindowClose, FaSadTear, FaFilter } from "react-icons/fa";
+import styles from "./styles.module.scss";
+import { useHistory } from "react-router-dom";
+
+export const Store = () => {
+  const {
+    getGames,
+    getGameCreators,
+    getDiscounts,
+    getUsedDiscounts,
+    getGenres,
+    getUsedGenres,
+  } = useApi();
+
+  const history = useHistory();
+  const formRef = useRef(null);
+
+  const [games, setGames] = useState<IGameForUI[]>([]);
+  const [gameCreators, setGameCreators] = useState<IGameCreator[]>([]);
+  const [genres, setGenres] = useState<IGenre[]>([]);
+
+  const [error, setError] = useState<IApiError | null>(null);
+  const [sortType, setSortType] = useState<"creationDate" | "alphabet">(
+    "alphabet"
+  );
+  const [filtersAmount, setFiltersAmount] = useState<number>(0);
+  const [filteredGames, setFilteredGames] = useState<IGameForUI[]>([]);
+  const [isFiltersMenuOpen, setIsFiltersMenuOpen] = useState<boolean>(false);
+
+  const removeFilters = () => {
+    if (formRef.current) {
+      const form = (formRef.current as any) as HTMLFormElement;
+      const formInputs = [...form.elements] as HTMLInputElement[];
+      formInputs.forEach((input) => input.checked && (input.checked = false));
+    }
+    setFilteredGames(sortGames(games, sortType));
+    setFiltersAmount(0);
+  };
+
+  const handleFilterTypeChange = (e: React.FormEvent) => {
+    const form = e.currentTarget as HTMLFormElement;
+    const formInputs = [...form.elements] as HTMLInputElement[];
+    const checkedFormInputs = formInputs.filter((input) => input.checked);
+
+    if (checkedFormInputs.length) {
```

Comment:
Always check `.length` with exact numeric operators. For example "is equal zero", "is larger than zero" and etc.